src/systemctl/systemctl-edit.c

```diff
@@ -122,25 +126,74 @@ static int create_edit_temp_file(const char *new_path, const char *original_path
         if (r < 0)
                 return log_error_errno(r, "Failed to create directories for \"%s\": %m", new_path);
 
-        r = mac_selinux_create_file_prepare(original_path, S_IFREG);
-        if (r < 0)
-                return r;
+        if (original_path) {
+                r = mac_selinux_create_file_prepare(new_path, S_IFREG);
+                if (r < 0)
+                        return r;
 
-        r = copy_file(original_path, t, 0, 0644, 0, 0, COPY_REFLINK);
-        if (r == -ENOENT) {
+                r = copy_file(original_path, t, 0, 0644, 0, 0, COPY_REFLINK);
+                if (r == -ENOENT) {
+                        r = touch(t);
+                        mac_selinux_create_file_clear();
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to create temporary file \"%s\": %m", t);
+                } else {
+                        mac_selinux_create_file_clear();
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to create temporary file for \"%s\": %m", new_path);
+                }
+        } else if (original_unit_paths) {
+                _cleanup_free_ char *new_contents = NULL;
+                _cleanup_fclose_ FILE *f = NULL;
+                char **path;
+                size_t size;
 
-                r = touch(t);
+                r = mac_selinux_create_file_prepare(new_path, S_IFREG);
+                if (r < 0)
+                        return r;
 
+                f = fopen(t, "we");
                 mac_selinux_create_file_clear();
+                if (!f)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", t);
 
+                r = fchmod(fileno(f), 0644);
                 if (r < 0)
-                        return log_error_errno(r, "Failed to create temporary file \"%s\": %m", t);
-
-        } else {
-                mac_selinux_create_file_clear();
+                        return log_error_errno(errno, "Failed to change mode of \"%s\": %m", t);
+
+                r = read_full_file(new_path, &new_contents, &size);
```

Comment:
hmm, maybe i am confused, but shouldn#t this be `original_path` you read from here?

Reply:
`new_path` is the path of the override file, e.g. `/etc/systemd/system/<service>.service.d/override.conf`. So this is reading and pasting the existing contents of the override file.

`original_path`: original unit that's being edited, only passed in with `systemctl edit -l`, e.g. `/usr/lib/systemd/system/<service>.service`  
`original_unit_paths`: all unit files that'd be listed out if you did a `systemctl cat`, only passed in with `systemctl edit`